Mathlib/Algebra/QuadraticAlgebra/Basic.lean

```diff
@@ -0,0 +1,389 @@
+/-
+Copyright (c) 2025 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir
+-/
+
+import Mathlib.Algebra.QuadraticAlgebra.Defs
+import Mathlib.Algebra.Star.Unitary
+import Mathlib.Tactic.FieldSimp
+
+/-! # Quadratic algebras : involution and norm.
+
+Let `R` be a commutative ring. We define:
+
+* `QuadraticAlgebra.star` : the quadratic involution
+
+* `QuadraticAlgebra.norm` : the norm
+
+We prove :
+
+* `QuadraticAlgebra.isUnit_iff_norm_isUnit`:
+  `w : QuadraticAlgebra R a b` is a unit iff `w.norm` is a unit in `R`.
+
+* `QuadraticAlgebra.norm_mem_nonZero_divisors_iff`:
+  `w : QuadraticAlgebra R a b` isn't a zero divisor iff
+  `w.norm` isn't a zero divisor in `R`.
+
+* If `R` is a field, and `∀ r, r ^ 2 ≠ a + b * r`, then `QuadraticAlgebra R a b` is a field.
+`
+-/
+
+namespace QuadraticAlgebra
+
+variable {R : Type*} {a b : R}
+
+section omega
+
+section
+
+variable [Zero R] [One R]
+
+/-- The representative of the root in the quadratic algebra -/
+def ω : QuadraticAlgebra R a b :=
```

Comment:
I think the declaration *name* should be `ω`, and you should introduced scoped notation `ω` for the definition.
```suggestion
def omega : QuadraticAlgebra R a b :=
```
This aligns with your naming below (e.g., `re_omega`)